app.js

```diff
@@ -34,26 +34,25 @@ class NetworkSpeedCheck {
     });
   }
 
-  checkUploadSpeed(options) {
-    const dataSizeInKb = 20;
+  checkUploadSpeed(options, fileSizeInBytes = 2000000) {
     let startTime;
-    const data = '{"data": "' + this.generateTestData(20) + '"}';
+    const data = '{"data": "' + this.generateTestData(fileSizeInBytes / 1000) + '"}';
```

Comment:
I believe codacy is failing because of double quotes for strings. Switching from `'{"data": "' + this.generateTestData(fileSizeInBytes / 1000) + '"}'` to `"{"data": "' + this.generateTestData(fileSizeInBytes / 1000) + '"}"` should fix it